src/reducers/pieces.js

```diff
@@ -3,11 +3,24 @@ import C from '../constants'
 const piece = (piece = {}, action) => {
     switch (action.type) {
         case C.PIECE_UPDATE:
-            return {...piece, ...action.piece, changed: !(action.notChanged || (action.piece.data.html === piece.data.html)) || piece.changed};
+            return {
+                ...piece, ...action.piece,
+                changed: !(action.notChanged || (action.piece.data.html === piece.data.html)) || piece.changed
+            };
         case C.PIECE_RESET:
             return {...piece, changed: false};
         case C.PIECE_REMOVE:
             return {...piece, destroy: true};
+        case C.PIECE_SET_DATA:
+            //check is initiated
+            if(!piece.fetched){
+                console.error(`Piece was not initialized before use setDate function. Piece id: ${piece.id}`);
```

Comment:
Additionally add a check in setData public method and produce an error with `throw new Error` there if there is no piece data or no 'byId' at all